src/basic-samples/GuaranteedPublisher.js

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Solace Web Messaging API for JavaScript
+ * Publishing Guaranteed messages on a Topic tutorial - Guaranteed publisher
+ * Demonstrates sending persistent messages on a topic
+ */
+
+/*jslint es6 browser devel:true*/
+/*global solace*/
+
+var GuaranteedPublisher = function (solaceModule, topicName) {
```

Comment:
`solaceModule` is not used in the function 🤔  we need to add `var solace = solaceModule;` since the solace module is already passed upon instantiating the function 
`var publisher = new GuaranteedPublisher(solace, 'solace/samples/nodejs/pers');`

Reply:
Done